core/src/test/scala/unit/kafka/cluster/BrokerEndPointTest.scala

```diff
@@ -1,204 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one or more
- * contributor license agreements.  See the NOTICE file distributed with
- * this work for additional information regarding copyright ownership.
- * The ASF licenses this file to You under the Apache License, Version 2.0
- * (the "License"); you may not use this file except in compliance with
- * the License.  You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package kafka.cluster
-
-import java.nio.charset.StandardCharsets
-import kafka.zk.BrokerIdZNode
-import org.apache.kafka.common.feature.{Features, SupportedVersionRange}
-import org.apache.kafka.common.feature.Features._
-import org.apache.kafka.common.network.ListenerName
-import org.apache.kafka.common.security.auth.SecurityProtocol
-import org.apache.kafka.server.network.BrokerEndPoint
-import org.junit.jupiter.api.Assertions.{assertEquals, assertNotEquals}
-import org.junit.jupiter.api.Test
-
-import scala.jdk.CollectionConverters._
-
-class BrokerEndPointTest {
-
-  @Test
-  def testHashAndEquals(): Unit = {
-    val broker1 = new BrokerEndPoint(1, "myhost", 9092)
-    val broker2 = new BrokerEndPoint(1, "myhost", 9092)
-    val broker3 = new BrokerEndPoint(2, "myhost", 1111)
-    val broker4 = new BrokerEndPoint(1, "other", 1111)
-
-    assertEquals(broker1, broker2)
-    assertNotEquals(broker1, broker3)
-    assertNotEquals(broker1, broker4)
-    assertEquals(broker1.hashCode, broker2.hashCode)
-    assertNotEquals(broker1.hashCode, broker3.hashCode)
-    assertNotEquals(broker1.hashCode, broker4.hashCode)
-
-    assertEquals(Some(1), Map(broker1 -> 1).get(broker1))
```

Comment:
I guess this test is unrelated to the zk stuff, but not too useful if we convert `BrookerEndPoint` to a record. I submitted #18577 for the latter.